### ISSUE TITLE
No provider for AnimationEngine in Universal with 4.4.1

### ISSUE THREAD

**Issue by hansl** [angular/angular#19250]
See https://travis-ci.org/angular/angular-cli/jobs/276772334#L1374

The error message reported at runtime is:

```
    ERROR { Error: No provider for AnimationEngine!
        at Error (native)
        at injectionError (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:1201:90)
        at noProviderError (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:1239:12)
        at ReflectiveInjector_._throwOrNull (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2681:19)
        at ReflectiveInjector_._getByKeyDefault (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2720:25)
        at ReflectiveInjector_._getByKey (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2652:25)
        at ReflectiveInjector_.get (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2521:21)
        at resolveNgModuleDep (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9521:25)
        at _callFactory (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9599:73)
        at _createProviderInstance$1 (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9535:26)
      addKey: [Function: addKey],
      keys: [ ReflectiveKey { token: [Function: AnimationEngine], id: 53 } ],
      injectors: 
       [ ReflectiveInjector_ {
           _constructionCounter: 1,
           _providers: [Object],
           _parent: [Object],
           keyIds: [Object],
           objs: [Object] } ],
      constructResolvingMessage: [Function],
      ngOriginalError: undefined,
      __zone_symbol__currentTask: 
       ZoneTask {
         _zone: 
          Zone {
            _properties: [Object],
            _parent: [Object],
            _name: 'angular',
            _zoneDelegate: [Object] },
         runCount: 0,
         _zoneDelegates: null,
         _state: 'notScheduled',
         type: 'microTask',
         source: 'Promise.then',
         data: undefined,
         scheduleFn: undefined,
         cancelFn: null,
         callback: [Function],
         invoke: [Function] } }
    Unhandled Promise rejection: No provider for AnimationEngine! ; Zone: <root> ; Task: Promise.then ; Value: { Error: No provider for AnimationEngine!
        at Error (native)
        at injectionError (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:1201:90)
        at noProviderError (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:1239:12)
        at ReflectiveInjector_._throwOrNull (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2681:19)
        at ReflectiveInjector_._getByKeyDefault (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2720:25)
        at ReflectiveInjector_._getByKey (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2652:25)
        at ReflectiveInjector_.get (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2521:21)
        at resolveNgModuleDep (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9521:25)
        at _callFactory (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9599:73)
        at _createProviderInstance$1 (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9535:26)
      addKey: [Function: addKey],
      keys: [ ReflectiveKey { token: [Function: AnimationEngine], id: 53 } ],
      injectors: 
       [ ReflectiveInjector_ {
           _constructionCounter: 1,
           _providers: [Object],
           _parent: [Object],
           keyIds: [Object],
           objs: [Object] } ],
      constructResolvingMessage: [Function],
      ngOriginalError: undefined,
      __zone_symbol__currentTask: 
       ZoneTask {
         _zone: 
          Zone {
            _properties: {},
            _parent: null,
            _name: '<root>',
            _zoneDelegate: [Object] },
         runCount: 0,
         _zoneDelegates: null,
         _state: 'notScheduled',
         type: 'microTask',
         source: 'Promise.then',
         data: undefined,
         scheduleFn: undefined,
         cancelFn: null,
         callback: [Function],
         invoke: [Function] } } Error: No provider for AnimationEngine!
        at Error (native)
        at injectionError (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:1201:90)
        at noProviderError (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:1239:12)
        at ReflectiveInjector_._throwOrNull (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2681:19)
        at ReflectiveInjector_._getByKeyDefault (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2720:25)
        at ReflectiveInjector_._getByKey (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2652:25)
        at ReflectiveInjector_.get (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:2521:21)
        at resolveNgModuleDep (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9521:25)
        at _callFactory (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9599:73)
        at _createProviderInstance$1 (/tmp/angular-cli-e2e-117818-4693-1ghv9w2.k2pqt9y66r/test-project/node_modules/@angular/core/bundles/core.umd.js:9535:26)
```

Repro:

1. clone https://github.com/hansl/platform-server-issue-repro
2. `npm install`
2. `ng build`
3. `node index.js`

The expected result is no errors (a console message will be outputted saying Angular is running in dev mode).

The actual result is the error above.

**Comment by almothafar** [comment]
Also getting No provider issue for `ElementRef` for AoT build.

**Comment by MichaelSolati** [comment]
Oh thank god, I thought I was crazy when this started happening.

**Comment by IgorMinar** [comment]
this issue was resolved in the 4.4.2 release

**Comment by SamuelMarks** [comment]
@MichaelSolati: Ditto! - 😥 🤒 

My build was failing and I kept scratching my head wondering what I did wrong. So I guess the compiler can be wrong 🤣
